From 7013bbe99f0b159f58c47ac994b68cd6db9bae9d Mon Sep 17 00:00:00 2001
From: Tom Gundersen <teg@jklm.no>
Date: Wed, 10 Sep 2014 10:56:26 +0200
Subject: [PATCH] udev: timeout - increase timeout

Some kernel modules still take more than one minute to insmod, we no longer rely on the timeout
killing insmod within a given period of time, so just bump this to a much higher value. Its only
purpose is to make sure that nothing stays aronud forever.

(cherry picked from commit b5338a19864ac3f5632aee48069a669479621dca)

Conflicts:
	src/udev/udevd.c

[zj: This commit is not really similar to b533... at all, but hopefuly the result is alike.]
---
 src/udev/udevd.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/udev/udevd.c b/src/udev/udevd.c
index bbca5d5bb4..c5dd6de31a 100644
--- a/src/udev/udevd.c
+++ b/src/udev/udevd.c
@@ -1240,7 +1240,7 @@ int main(int argc, char *argv[])
                                 break;
 
                         /* timeout at exit for workers to finish */
-                        timeout = 30 * 1000;
+                        timeout = 180 * 1000;
                 } else if (udev_list_node_is_empty(&event_list) && !children) {
                         /* we are idle */
                         timeout = -1;
@@ -1278,7 +1278,7 @@ int main(int argc, char *argv[])
                                 if (worker->state != WORKER_RUNNING)
                                         continue;
 
-                                if ((now(CLOCK_MONOTONIC) - worker->event_start_usec) > 30 * 1000 * 1000) {
+                                if ((now(CLOCK_MONOTONIC) - worker->event_start_usec) > 180 * 1000 * 1000) {
                                         log_error("worker [%u] %s timeout; kill it\n", worker->pid,
                                             worker->event ? worker->event->devpath : "<idle>");
                                         kill(worker->pid, SIGKILL);
